src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -773,7 +785,8 @@ private class FastqDemultiplexer(val sampleInfos: Seq[SampleInfo],
       }
 
     val passQc = if (!keepOnlyPassing) true else demuxRecords.forall(d => d.readInfo.forall(_.passQc))
-    DemuxResult(sampleInfo=sampleInfo, numMismatches=numMismatches, records=demuxRecords, passQc=passQc)
+    val isControl = if (keepControlReads) false else demuxRecords.forall(d => d.readInfo.forall(_.internalControl))
```

Comment:
ditto to this comment: https://github.com/fulcrumgenomics/fgbio/pull/713/files#r712552311.  I think we want to set the value regardless of `keepControlReads`, then filter later (in the `resultsIterator.filter`)
```suggestion
    val isControl = demuxRecords.forall(d => d.readInfo.forall(_.internalControl))
```